database/challenge.go

```diff
@@ -119,8 +133,15 @@ func (db *DB) ValidateChallengeResponse(ctx context.Context, chr ChallengeRespon
 	// Now that we know the challenge type, we can get the recipient as well.
 	recipientOffset := ChallengeSize + len([]byte(cType))
 	recipient := string(resp[recipientOffset:])
-	if recipient != PortalName {
-		return nil, primitive.ObjectID{}, errors.New("invalid recipient " + recipient)
+	// Make sure the recipient has a schema. It usually won't. We check for
+	// `http` only in order to cover both `http://` and `https://`
+	if !strings.HasPrefix(recipient, "http") {
```

Comment:
`nit` do we care that this doesn't work when the recipient is `httpsomethingsomething`?

Reply:
It works now.